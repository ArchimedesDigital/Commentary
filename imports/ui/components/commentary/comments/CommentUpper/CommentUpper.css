.comment-upper {
    display: block;
    width: 90%;
    margin: 0 auto;
    clear: both;
    padding-top: 50px;
}

.comment-upper-left {
    display: inline-block;
    vertical-align: top;
}

.comment-upper-right {
    display: inline-block;
    vertical-align: top;
    text-align: right;
    float: right;
}

.comment-author {
    margin: 12px 0 0;
}

.comment-author-text {
    display: inline-block;
    vertical-align: top;
    padding: 4px 10px;
}

.comment-author-image-wrap {
    display: inline-block;
    vertical-align: top;
    border-radius: 50px;
    width: 50px;
    height: 50px;
    overflow: hidden;
    border: 1px solid #b6b6b6;
}

.comment-author-name {
    display: block;
    font-size: 16px;
}

.comment-author-name:hover {
    color: #aaa;
}
