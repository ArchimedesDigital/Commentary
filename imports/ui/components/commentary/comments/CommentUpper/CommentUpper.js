import AvatarIcon from '/imports/avatar/client/ui/AvatarIcon.jsx';  // eslint-disable-line import/no-absolute-path
import FontIcon from 'material-ui/FontIcon';


/*
	BEGIN CommentUpperLeft 
*/
const CommentUpperLeft = props => (
	<div className="comment-upper-left">
		<h1>{props.title}</h1>
	</div>
);
CommentUpperLeft.propTypes = {
	title: React.PropTypes.string.isRequired,
};
/*
	END CommentUpperLeft 
*/



/*
	BEGIN CommentUpperRight 
*/
const CommentUpperRight = props => (
	<div className="comment-upper-right">
		{props.commenters.map(commenter => (
			<div
				key={commenter._id}
				className="comment-author"
			>
				{props.userCanEditCommenters.indexOf(commenter._id) > -1 ?
					<FlatButton
						label="Edit comment"
						href={`/commentary/${comment._id}/edit`}
						icon={<FontIcon className="mdi mdi-pen" />}
					/>
					:
					''
				}
				<div className={"comment-author-text"}>
					<a href={`/commenters/${commenter.slug}`}>
						<span className="comment-author-name">{commenter.name}</span>
					</a>
					<span>
						{props.updateDate}
					</span>
				</div>
				<div className="comment-author-image-wrap paper-shadow">
					<a href={`/commenters/${commenter.slug}`}>
						<AvatarIcon
							avatar={
								(commenter && 'avatar' in commenter) ?
								commenter.avatar.src
								: null
							}
						/>
					</a>
				</div>
			</div>
		))}
	</div>
);
CommentUpperRight.propTypes = {
	commenters: React.PropTypes.arrayOf(React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		slug: React.PropTypes.string.isRequired,
		name: React.PropTypes.string.isRequired,
		avatar: React.PropTypes.shape({
			src: React.PropTypes.string.isRequired,
		})
	})).isRequired,
	updateDate: React.PropTypes.string.isRequired,
	userCanEditCommenters: React.PropTypes.arrayOf(React.PropTypes.string),
};
CommentUpperRight.defaultProps = {
	userCanEditCommenters: [],
};
/*
	END CommentUpperRight 
*/



/*
	BEGIN CommentUpper 
*/
const CommentUpper = (props) => (
	<div className="comment-upper">
		{!props.hideTitle && <CommentUpperLeft
			title={props.title}
		/>}
		{!props.hideCommenters && <CommentUpperRight
			commenters={props.commenters}
			updateDate={props.updateDate}
		/>}
	</div>
);
CommentUpper.propTypes = {
	title: React.PropTypes.string.isRequired,
	commenters: React.PropTypes.arrayOf(React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		slug: React.PropTypes.string.isRequired,
		name: React.PropTypes.string.isRequired,
		avatar: React.PropTypes.shape({
			src: React.PropTypes.string.isRequired,
		})
	})).isRequired,
	updateDate: React.PropTypes.string.isRequired,
	userCanEditCommenters: React.PropTypes.arrayOf(React.PropTypes.string),
	hideTitle: React.PropTypes.bool,
	hideCommenters: React.PropTypes.bool,
};
CommentUpper.defaultProps = {
	userCanEditCommenters: [],
	hideTitle: false,
	hideCommenters: false,
};
/*
	END CommentUpper 
*/

export default CommentUpper;
export { CommentUpperLeft, CommentUpperRight };
