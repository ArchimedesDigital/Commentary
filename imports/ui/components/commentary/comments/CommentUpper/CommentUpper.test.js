import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentUpper from './CommentUpper';

jest.mock('material-ui/FlatButton/FlatButton');

describe('CommentUpper', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentUpper title="" commentId="" commenters={[]} updateDate="" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
