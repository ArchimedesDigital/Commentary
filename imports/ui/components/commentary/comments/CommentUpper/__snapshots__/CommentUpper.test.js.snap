// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentUpper renders correctly 1`] = `
<div
  className="comment-upper"
>
  <div
    className="comment-upper-left"
  >
    <h1>
      
    </h1>
  </div>
  <div
    className="comment-upper-right"
  />
</div>
`;
